analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/UtilityAnalyzerBaseTest.cs

```diff
@@ -136,12 +135,19 @@ public TestUtilityAnalyzer(string language, params string[] additionalPaths) : b
                     LanguageNames.VisualBasic => VisualBasicCompilation.Create(null),
                     _ => throw new InvalidOperationException($"Unexpected {nameof(language)}: {language}")
                 };
-                var c = new CompilationAnalysisContext(compilation, new AnalyzerOptions(additionalFiles), null, null, default);
-                ReadParameters(new(AnalysisScaffolding.CreateSonarAnalysisContext(), c));
+                var analyzerOptions = new AnalyzerOptions(additionalFiles);
+                var config = analyzerOptions.SonarProjectConfig() is { } sonarProjectConfig ? new ProjectConfigReader(sonarProjectConfig.GetText()) : ProjectConfigReader.Empty;
+                var isTestProject = config.ProjectType switch
+                {
+                    ProjectType.Unknown => compilation.IsTest(),
+                    ProjectType.Test => true,
+                    _ => false
+                };
```

Comment:
There must be a better way to solve this. `ProjectConfigReader` and the `IsTestProject` extension method are the two obstacles to solve here.

Reply:
Something like this should do the trick
```
                var context = new Mock<CompilationStartAnalysisContext>(compilation, new AnalyzerOptions(additionalFiles), default).Object;
                ReadParameters(new SonarCompilationStartAnalysisContext(AnalysisScaffolding.CreateSonarAnalysisContext(), context));
```

Reply:
Yep. That was it. Easy once you see it, but hard if you have to come up with this by yourself.

Reply:
Indeed. I also stole it from other UT for practical reasons :)